app/main.py

```diff
@@ -1,2 +1,55 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.capacity = 8
+        self.load_factor = 0.66
+        self.list = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key: (int, float, str, tuple, bool), value) -> None:
+        if len(self) + 1 > self.capacity * self.load_factor:
+            self.increase_capacity()
+
+        key_hash = hash(key) % self.capacity
+
+        if not self.list[key_hash]:  # check if cell is empty
+            self.list[key_hash] = [key, value, hash(key)]
+            return
+
+        for i in range(len(self.list)):  # check if we have key in self.list
+            try:
+                if self.list[i][0] == key:
+                    self.list[i][1] = value
+                    return
+            except IndexError:
+                pass
+
+        while True:  # find and fill next empty cell
+            if key_hash + 1 == self.capacity:
+                key_hash = 0
+            else:
+                key_hash += 1
+            if not self.list[key_hash]:
+                self.list[key_hash] = [key, value, hash(key)]
+                return
+
+    def __getitem__(self, key: (int, float, str, tuple, bool)) -> None:
+        for cell in self.list:
+            try:
+                if key == cell[0]:
+                    return cell[1]
+            except IndexError:
+                pass
+        raise KeyError
```

Comment:
It's not a very good practice to raise errors without messages. Python's dict uses the key it tried to find as message for `KeyError`. I think we should do the same.